dashboard/app/models/levels/blockly.rb

```diff
@@ -112,6 +112,7 @@ def uses_google_blockly?
   def summarize_for_lab2_properties(script, script_level = nil, current_user = nil)
     level_properties = super
     level_properties[:sharedBlocks] = localized_blockly_level_options(script)["sharedBlocks"]
+    localize_blockly_level_options_for_lab2(script)
```

Comment:
I think you mentioned this offline, but do we not need to assign this back to `level_data["startSources"]`?

Reply:
I think the reuse of the cache might explain why this happened to work.  Have now updated to something more deliberate.